### PR TITLE
Check kubeConfig before starting sandbox #minor

### PR DESCRIPTION
**Pull request by daniel-shuy**
# TL;DR
Adds a pre-check to verify if the local kubeconfig exists before starting the sandbox so that it fails-fast if the kubeconfig doesn't exist, instead of failing when trying to copy the sandbox context to the local kubecontext.

## Type
- [ ] Bug Fix
- [x] Feature
- [ ] Plugin

## Are all requirements met?

- [x] Code completed
- [x] Smoke tested
- [x] Unit tests added
- [x] Code documentation added
- [x] Any pending items have an associated Issue

## Complete description
Abstracted out the check from `ContextOps#CopyContext()` to a new `ContextOps#CheckConfig()` function. `StartCluster()` would then instantiate a new `ContextManager` and call `ContextManager#CheckConfig()` before starting the sandbox.

This required re-generating `pkg/k8s/mocks/context_ops.go` using `mockery`.
Running the latest version (`2.x.x`) of [mockery](https://github.com/vektra/mockery) with `mockery -name=ContextOps -case=underscore` fails because the `-name` option has been changed to `-n`/`--name`. Running with a `1.x.x` version of [mockery](https://github.com/vektra/mockery) fails with:
```
package "fmt" without types was imported from ... without types
```
Eventually I discovered that `flytectl` uses a fork of `mockery` (https://github.com/EngHabu/mockery), and that I could run [make generate](https://github.com/flyteorg/flytectl/blob/v0.6.18/boilerplate/flyte/golang_test_targets/Makefile#L11-L13) to install and run it. This really should be documented in the contribution guide (`docs/source/contribute.rst`).

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2943

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by pmahindrakar-oss** [comment]
@daniel-shuy looks good to me. 
Can you add to the description what error is shown to the user when the local kube context doesn't exist. Hope that message is self explanatory for people to create one or may be we can add an explicit log message for this scenario to use a kube client to initialise this

**Review by pmahindrakar-oss — APPROVED** [review]

**Comment by pmahindrakar-oss** [comment]
@daniel-shuy i verified your changes by changing permissions on the ~/.kube directory

New behavior.
```
➜  flytectl git:(feature/sandbox-precheck-local-kubecontext) bin/flytectl demo start
Error: error loading config file "/Users/praful/.kube/config": open /Users/praful/.kube/config: permission denied
```

Old behavior
```
flytectl demo start
🧑‍🏭 Bootstrapping a brand new flyte cluster... 🔨 🔧
cr.flyte.org/flyteorg/flyte-sandbox-lite:sha-4f73dc6994dfeafb9eecd9b17d16d7f9275b577a Fully Qualified image
🐋 Running Flyte v1.2.0 release
🐋 pulling docker image for release cr.flyte.org/flyteorg/flyte-sandbox-lite:sha-4f73dc6994dfeafb9eecd9b17d16d7f9275b577a
{"status":"Pulling from flyteorg/flyte-sandbox-lite","id":"sha-4f73dc6994dfeafb9eecd9b17d16d7f9275b577a"}
{"status":"Digest: sha256:482ad8fb1542e79e8561b62cd559ce787af0a268d7576babbe54c253d549c5cd"}
{"status":"Status: Image is up to date for cr.flyte.org/flyteorg/flyte-sandbox-lite:sha-4f73dc6994dfeafb9eecd9b17d16d7f9275b577a"}
🧑‍🏭 booting Flyte-sandbox container
92022-10-26T06:31:25.910671461Z Starting Docker daemon...
%2022-10-26T06:31:26.203130062Z Done.
72022-10-26T06:31:26.203198314Z Starting k3s cluster...
%2022-10-26T06:31:31.790252603Z Done.
Error: error loading config file "/Users/praful/.kube/config": open /Users/praful/.kube/config: permission denied
```

So its good we fail early on this and give the user the error on what should be done.
Thanks

**Comment by daniel-shuy** [comment]
@pmahindrakar-oss Great, thanks! I tried deleting my `~/.kube` directory but it just created a new one 😆

**Review by pmahindrakar-oss — APPROVED** [review]

**Comment by pmahindrakar-oss** [comment]
@daniel-shuy i think the validation should cover any load issues of any existing config aswell

**Comment by samhita-alla** [comment]
@daniel-shuy, fill out [this form](https://tally.so/r/nWO7qQ) to claim your hacktoberfest swag.
